### PR TITLE
feat: fetch index information for columns

### PR DESCRIPTION
**Pull request by aaqilniz**
This PR along with another PR@loopback-datasource-juggler will enable models to have index information.

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

File: test/mysql.discover.test.js [hunk 1]
```diff
@@ -199,6 +199,23 @@ describe('Discover model primary keys', function() {
   });
 });
 
+describe('Discover user model with index', function() {
```
Comment (samarpanB) [inline test/mysql.discover.test.js:202]: can you please add some more test cases to cover more negative scenarios and other worst cases ?

**Review by samarpanB — CHANGES_REQUESTED** [review]
